### ISSUE TITLE
[Bug] Parameter public check

### ISSUE THREAD

**Issue by Akeit0** [Cysharp/PrivateProxy#7]
In this part, parameters public checks do nothing. (foreach continue)

```cs
 else if (generateMethod && item is IMethodSymbol m)
  {
      // both return type and parameter type must be public
      if (m.ReturnType.DeclaredAccessibility != Accessibility.Public) continue;
      foreach (var parameter in m.Parameters)
      {
          if (parameter.Type.DeclaredAccessibility != Accessibility.Public) continue; 
      }

      if (m.DeclaredAccessibility == Accessibility.Public) continue;
      list.Add(new(item));
  }
```
And only checking Accessibility.Public doesn't care about element types.
Following is what I'm using in my library.
```cs
static bool CanExpose(ITypeSymbol typeSymbol)
{
    if (typeSymbol is IPointerTypeSymbol) return false;
    while(typeSymbol is IArrayTypeSymbol arrayType)
    {
        typeSymbol = arrayType.ElementType;
        if (typeSymbol is IPointerTypeSymbol) return false;
    }
    if(typeSymbol is INamedTypeSymbol namedTypeSymbol)
    {
        if (namedTypeSymbol.IsGenericType)
        {
            foreach (var e in namedTypeSymbol.TypeArguments)
            {
                if (!CanExpose(e)) return false;
            }
        }
           
    }
    var declaredAccessibility = typeSymbol.DeclaredAccessibility;
    switch (declaredAccessibility)
    {
        case Accessibility.Public:
            return true;
        case Accessibility.NotApplicable:
            return true;
        default:
            return false;
    }
}
```

**Comment by neuecc** [comment]
thanks, I'll look soon.

**Comment by hadashiA** [comment]
Hi. I have taken this suggestion into consideration. #9

NOTE:
BTW, should `NotApplicable` be true?
My understanding is that if Type is NotApplicable, it as the same as internal I think.
Any suggestions are welcome.

**Comment by Akeit0** [comment]
> Hi. I have taken this suggestion into consideration. #9
> 
> NOTE: BTW, should `NotApplicable` be true? My understanding is that if Type is NotApplicable, it as the same as internal I think. Any suggestions are welcome.

Originally, this was to accommodate Generic parameters.
(It' not sure but I think arrays were NotApplicable as well.)

**Comment by Akeit0** [comment]
As far as I have tried, internal is not Accessibility.NotApplicable.

**Comment by hadashiA** [comment]
Oh, Thanks for pointing that out. I'll investigate.

**Comment by hadashiA** [comment]
I have investigated.

- As you said, the Array was Accesibility.NotApplicable.
- Also, Accessibility.Internal both when I omitted the access modifier and when I explicitly specified internal. (I misunderstood the ISymbol abstraction ><)

Further fixes to #9.
Thanks a lot!
